jupyter_server/services/kernels/kernelmanager.py

```diff
@@ -179,10 +179,14 @@ async def start_kernel(self, kernel_id=None, path=None, **kwargs):
             The name identifying which kernel spec to launch. This is ignored if
             an existing kernel is returned, but it may be checked in the future.
         """
-        if kernel_id is None:
+        if kernel_id is None or kernel_id not in self:
             if path is not None:
                 kwargs['cwd'] = self.cwd_for_path(path)
-            kernel_id = await ensure_async(self.pinned_superclass.start_kernel(self, **kwargs))
+            if kernel_id:
+                kernel_id = await ensure_async(self.pinned_superclass.start_kernel(self, kernel_id=kernel_id, **kwargs))
```

Comment:
Could we take an approach similar to how 'cwd' is set into `kwargs` and retain the single call to `super.start_kernel()`?  That just seems a bit cleaner to me.

Reply:
absolutely, thanks for pointing that out. I did feel kind of repeating myself in the original code